### ISSUE TITLE
Julia: ASCIIString error

### ISSUE THREAD

**Issue by antoine-levitt** [emacs-ess/ESS#424]
Hi,

With (setq inferior-julia-args "-q") I get
ERROR: UndefVarError: ASCIIString not defined
when running julia with 0.6rc. ASCIIString was apparently deprecated, and "String" should be used now, and indeed making that change fixes the error. I don't understand why this doesn't error without "-q".

**Comment by vspinu** [comment]
PRs are welcome. Does this break older versions?

**Comment by antoine-levitt** [comment]
It shouldn't, but I don't know how to test, I don't know even what fun_args does... patch below in any case.
```
diff --git a/etc/ess-julia.jl b/etc/ess-julia.jl
index 313e847..4283e4e 100644
--- a/etc/ess-julia.jl
+++ b/etc/ess-julia.jl
@@ -47,10 +47,10 @@ function fun_args(f::Function)
     print("))")
 end
 
-function fun_args(s::ASCIIString)
+function fun_args(s::String)
     try
         m = eval(current_module(), parse(s))
-        if typeof(m) != ASCIIString
+        if typeof(m) != String
             fun_args(m)
         end
     catch

```

**Comment by nverno** [comment]
This should be fixed along with #442

**Comment by antoine-levitt** [comment]
Wonderful! Now I only get a slightly annoying

```
WARNING: Method definition function_module(Function) in module Base at reflection.jl:906 overwritten in module ESS at none:26.
ESS
```
which comes from 
`VERSION >= v"0.4-" && (Base.function_module(f::Function)=typeof(f).name.module)`
in ess-julia.jl. I commented that out and it didn't seem to break anything for me, any idea why this was done?

**Comment by nverno** [comment]
It had been defined differently by @vspinu back when functions were represented differently.  I think we can probably remove it?  v0.5 and v0.6 both appear to have `Base.function_module` that returns the module with the first definition of the function

**Comment by jabranham** [comment]
Closing since this was fixed by #443
